config/envoyconfig/clusters.go

```diff
@@ -201,6 +216,7 @@ func (b *Builder) buildPolicyTransportSocket(
 	options *config.Options,
 	policy *config.Policy,
 	dst url.URL,
+	upstreamProtocol upstreamProtocolConfig,
```

Comment:
Having an additional function argument for upstreamProtocol here feels weird to me since we already have the info we need in `policy *config.Policy` to determine the upstreamProtocol.